test/e2e/cronjob.go

```diff
@@ -225,10 +235,227 @@ var _ = framework.KubeDescribe("CronJob", func() {
 		err = deleteCronJob(f.ClientSet, f.Namespace.Name, cronJob.Name)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	// cleanup of successful finished jobs, with default limits
+	It("should delete successful finished jobs according to default history limits [Slow]", func() {
```

Comment:
Appreciate the amount of e2e tests, but @spxtr is trying to decrease the time they run (see https://github.com/kubernetes/kubernetes/issues/40839 for discussion). Your new tests will significantly increase that number, unfortunately. I'd suggest one test with limits set to 1 and 1 and verify it's working as expected, the rest has already pretty good coverage in unit tests.